sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -153,7 +153,8 @@ object HiveAnalysis extends Rule[LogicalPlan] {
     case CreateTable(tableDesc, mode, None) if DDLUtils.isHiveTable(tableDesc) =>
       CreateTableCommand(tableDesc, ignoreIfExists = mode == SaveMode.Ignore)
 
-    case CreateTable(tableDesc, mode, Some(query)) if DDLUtils.isHiveTable(tableDesc) =>
+    case CreateTable(tableDesc, mode, Some(query))
+        if DDLUtils.isHiveTable(tableDesc) && query.resolved =>
```

Comment:
This sounds like a separate issue. Could we submit a separate PR?

Reply:
This PR prevents lookupTableFromCatalog from throwing NoSuchTableException right away. Instead, it relies on checkAnalysis to throw an exception for UnresolvedRelation.

The test hive.SQLQuerySuite."double nested data" would fail in the following sql without this change:
```
CREATE TABLE test_ctas_1234 AS SELECT * from notexists
```
HiveAnalysis gets to run before checkAnalysis, thus exposing this bug where query.output is used before query is resolved.

So wouldn't say it is a totally separate issue.
In addition, outside of this PR, it'd be hard to write a unit test.

Reply:
I agree with John. This is needed as a consequence of fixing the `ResolveRelations` rule to no longer throw `AnalysisException` if it can't resolve the name and doesn't think that `ResolveSQLOnFile` would either.